network/http/webcheck/detectors-webcheck.tf

```diff
@@ -0,0 +1,225 @@
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+}
+
+resource "signalfx_detector" "tls_certificate_expiration" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] TLS certificate expiring count"
+
+	program_text = <<-EOF
+		signal = data('http.cert_expiry', ${module.filter-tags.filter_custom}).below(${var.tls_certificate_expiration_timeframe}, inclusive=True).count(by=['server_name', 'url'])${var.tls_certificate_expiration_aggregation_function}.${var.tls_certificate_expiration_transformation_function}(over='${var.tls_certificate_expiration_transformation_window}').publish('signal')
+		detect(when(signal > ${var.tls_certificate_expiration_threshold_critical})).publish('CRIT')
+		detect(when(signal > ${var.tls_certificate_expiration_threshold_warning}) and when(signal <= ${var.tls_certificate_expiration_threshold_critical})).publish('WARN')
+	EOF
+
+	rule {
+		description           = "is too high > ${var.tls_certificate_expiration_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.tls_certificate_expiration_disabled_critical, var.tls_certificate_expiration_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.tls_certificate_expiration_notifications_critical, var.tls_certificate_expiration_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "is too high > ${var.tls_certificate_expiration_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.tls_certificate_expiration_disabled_warning, var.tls_certificate_expiration_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.tls_certificate_expiration_notifications_warning, var.tls_certificate_expiration_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+}
+
+resource "signalfx_detector" "invalid_tls_certificate" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] TLS invalid certificate count"
+
+	program_text = <<-EOF
+		signal = data('http.cert_valid', ${module.filter-tags.filter_custom}).count()${var.invalid_tls_certificate_aggregation_function}.${var.invalid_tls_certificate_transformation_function}(over='${var.invalid_tls_certificate_transformation_window}').publish('signal')
+		detect(when(signal < ${var.invalid_tls_certificate_threshold_critical})).publish('CRIT')
+	EOF
+
+	rule {
+		description           = " < ${var.invalid_tls_certificate_threshold_critical}"
```

Comment:
hardcode value and remove variable using a message like `TLS Certificate is not valid`

Reply:
updated